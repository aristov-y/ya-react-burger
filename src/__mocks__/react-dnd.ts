const mockDropRef = jest.fn();
const mockDragRef = jest.fn();

const useDrag = jest.fn().mockImplementation(() => [undefined, mockDragRef]);
const useDrop = jest.fn().mockImplementation(() => [undefined, mockDropRef]);

export {
  useDrag,
  useDrop
}
