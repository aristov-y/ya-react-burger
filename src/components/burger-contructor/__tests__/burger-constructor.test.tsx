import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import BurgerConstructor from '../burger-constructor';
import { useSelector } from 'react-redux';
import { useDrop } from 'react-dnd';

jest.mock('../../order-details/order-details', () => () => 'OrderDetails');
jest.mock('../../../services/store');
jest.mock('../../../utils/get-order', () => Promise.resolve({}))
jest.mock('../../burger-constructor-element/burger-constructor-item');

const refs = {};
const getRef = (id) => {
  if (!refs[id]) {
    refs[id] = React.createRef();
  }
  return refs[id];
}

beforeEach(() => {
  const bun = {
    _id: 1,
    price: 2,
    name: 'Bun 01',
    image: 'Bun Image Source'
  };
  const main = [{
    _id: 1,
    price: 2,
    name: 'Bun 01'
  }];
  const store = {
    constructor: {
      bun,
      main
    },
    auth: {
      user: {
        name: 'test name'
      }
    }
  };
  (useSelector as jest.Mock)
    .mockImplementation((fn) => fn(store));
  (useDrop as jest.Mock)
    .mockImplementation((arg) => [undefined, getRef(arg.accept)]);
});

describe('burger-constructor', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<BurgerConstructor />);
    });

    expect(component.toJSON()).toMatchSnapshot();

    expect(useDrop).toBeCalledTimes(4);
  });
});
