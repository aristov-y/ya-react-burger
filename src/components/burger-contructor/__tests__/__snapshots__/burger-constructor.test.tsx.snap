// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`burger-constructor should render 1`] = `
<div
  className="mt-25"
  style={
    Object {
      "flex": "1",
      "maxWidth": "592px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "gap": "10px",
      }
    }
  >
    <div
      className="burger-constructor-item"
      data-cy="constructor-bun-top"
    >
      ConstructorElement
    </div>
    <div
      className="burger-constructor-scroller"
    >
      BurgerConstructorItem
    </div>
    <div
      className="burger-constructor-item"
      data-cy="constructor-bun-bottom"
    >
      ConstructorElement
    </div>
  </div>
  <div
    className="mt-10 burger-constructor-checkout"
    data-cy="checkout-block"
  >
    <span
      className="text text_type_digits-medium mr-10"
    >
      6
      CurrencyIcon
    </span>
    Button
  </div>
</div>
`;
